itest/lnd_coop_close_with_htlcs_test.go

```diff
@@ -101,3 +121,130 @@ func testCoopCloseWithHtlcs(ht *lntest.HarnessTest) {
+	// for it.
+	req := &routerrpc.SendPaymentRequest{
+		PaymentRequest: resp.PaymentRequest,
+		TimeoutSeconds: 60,
+		FeeLimitSat:    1000000,
+	}
+	ht.SendPaymentAndAssertStatus(bob, req, lnrpc.Payment_IN_FLIGHT)
+	ht.AssertNumActiveHtlcs(bob, 1)
+
+	// Assert at this point that the HTLC is open but not yet settled.
+	ht.AssertInvoiceState(invoiceStream, lnrpc.Invoice_ACCEPTED)
+
+	// We will now let Alice initiate the closure of the channel. We will
+	// also let her specify a specific delivery address to be used since we
+	// want to test that this same address is used in the Shutdown message
+	// on reconnection.
+	newAddr := alice.RPC.NewAddress(&lnrpc.NewAddressRequest{
+		Type: AddrTypeWitnessPubkeyHash,
+	})
+
+	_ = alice.RPC.CloseChannel(&lnrpc.CloseChannelRequest{
+		ChannelPoint:    chanPoint,
+		NoWait:          true,
+		DeliveryAddress: newAddr.Address,
+	})
+
+	// Assert that both nodes now see this channel as inactive.
+	ht.AssertChannelInactive(bob, chanPoint)
+	ht.AssertChannelInactive(alice, chanPoint)
+
+	// Now restart Alice and Bob.
+	ht.RestartNode(alice)
+	ht.RestartNode(bob)
+
+	ht.AssertConnected(alice, bob)
+
+	// Show that the channel is seen as active again by Alice and Bob.
+	//
+	// NOTE: This is a bug and will be fixed in an upcoming commit.
+	ht.AssertChannelActive(alice, chanPoint)
+	ht.AssertChannelActive(bob, chanPoint)
+
+	// Let's settle the invoice.
+	alice.RPC.SettleInvoice(preimage[:])
+
+	// Wait for the channel to appear in the waiting closed list.
+	//
+	// NOTE: this will time out at the moment since there is a bug that
+	// results in shutdown not properly being re-started after a reconnect.
+	err := wait.Predicate(func() bool {
+		pendingChansResp := alice.RPC.PendingChannels()
+		waitingClosed := pendingChansResp.WaitingCloseChannels
+
+		return len(waitingClosed) == 1
+	}, defaultTimeout)
+
+	// We assert here that there is a timeout error. This will be fixed in
+	// an upcoming commit.
+	require.Error(ht, err)
+
+	// Since the channel closure did not continue, we need to re-init the
+	// close.
+	closingTXID := ht.CloseChannel(alice, chanPoint)
+
+	// To further demonstrate the extent of the bug, we inspect the closing
+	// transaction here to show that the delivery address that Alice
+	// specified in her original close request is not the one that ended up
+	// being used.
+	//
+	// NOTE: this is a bug that will be fixed in an upcoming commit.
+	tx := alice.RPC.GetTransaction(&walletrpc.GetTransactionRequest{
+		Txid: closingTXID.String(),
+	})
+	require.Len(ht, tx.OutputDetails, 2)
+
+	// Find Alice's output in the coop-close transaction.
+	var outputDetail *lnrpc.OutputDetail
+	for _, output := range tx.OutputDetails {
+		if output.IsOurAddress {
+			outputDetail = output
+			break
+		}
+	}
+	require.NotNil(ht, outputDetail)
+
+	// Show that the address used is not the one she requested.
+	require.NotEqual(ht, outputDetail.Address, newAddr.Address)
+}
```

Comment:
@ProofOfKeags - continuing the thread here. 

Another thing I like about the approach of having the test pass is I can assert the incorrectness of multiple things. For example, here in this test: i test 1) that the nodes think the channel is still active & 2) that the node uses the incorrect address. 

> without realizing it then you have an "anti-test" lurking

I get the hesitation to add a passing test but I think that since it is heavily commented with "NOTE: bug", it's ok?

Reply:
You can do multiple failing assertions too. They do bury each other so that the first assertion to fail aborts the test but the point is that if you fix the code under test and it passes all of the failing assertions, then you're good.

I'm not gonna NACK merging this, though. I don't prefer it but I'm willing to accept it. The important thing is that by the end of the commit set, we have a good result.